pennylane/data/dataset.py

```diff
@@ -151,5 +183,40 @@ def from_dataset(cls, dataset):
             >>> print(vars(new_dataset))
             {'dtype': None, '__doc__': '', 'kw1': 1, 'kw2': '2', 'kw3': [3]}
         """
-        kwargs = {key: val for (key, val) in dataset.__dict__.items() if key not in ["dtype"]}
-        return cls(dtype=dataset.dtype, **kwargs)
+        return cls(**dataset.attrs)
+
+    # The methods below are intended for use only by standard Dataset objects
+    def __get_filename_for_attribute(self, attribute):
+        return self._fullfile if self._fullfile else self._prefix.format(attribute)
+
+    def __getattribute__(self, name):
+        try:
+            return super().__getattribute__(name)
+        except AttributeError as attr_error:
+            if not self._is_standard:
+                raise attr_error
+            filepath = self.__get_filename_for_attribute(name)
+            if os.path.exists(filepath):
+                # TODO: setattr?
+                filedata = self._read_file(filepath)
+                if name in filedata:
+                    return filedata[name]
+            # TODO: download the file here?
+            raise attr_error
+
+    def setdocstr(self, docstr, args=None, argtypes=None, argsdocs=None):
+        """Build the docstring for the Dataset class"""
+        docstring = f"{docstr}\n\n"
+        if args and argsdocs and argtypes:
+            docstring += "Args:\n"
+            for arg, argdoc, argtype in zip(args, argsdocs, argtypes):
+                docstring += f"\t{arg} ({argtype}): {argdoc}\n"
+            docstring += f"\nReturns:\n\tDataset({self._dtype})"
+        self.__doc__ = docstring  # pylint:disable=attribute-defined-outside-init
+
+
+def from_file(filepath):
```

Comment:
not exactly, Diego added this to what was then utils.py under the name `read()` but I felt this name and place is more fitting and pythonic. imho this is not entirely necessary because the user can just do this themselves, but I'm ok to offer this.